src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -207,7 +207,7 @@ CHIP_ERROR BindingManager::NotifyBoundClusterChanged(EndpointId endpoint, Cluste
                 else
                 {
                     mPendingNotificationMap.AddPendingNotification(iter.GetIndex(), context);
-                    if (peerDevice == nullptr || !peerDevice->IsConnecting())
+                    if (peerDevice == nullptr || !peerDevice->IsConnected())
```

Comment:
And what if the device is already connect**ing**? Maybe in both cases we shouldn't start establishing the connection again?

Reply:
The underlying `OperationalyDevice` class will not start a new connection if `Connect` is called when it's already connecting and enqueue the callback. Calling `EstablishConnection` again will be okay.